firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/chunks.txt

```diff
@@ -0,0 +1,282 @@
+
+Scenario: A ChunkSplitter correctly reports if input is overwritten before or after the output is consumed
+
+  GIVEN("A uint8_t chunksplitter constructed with default parameters with an empty internal buffer of capacity 256 bytes")
+
+    WHEN("input to the chunksplitter is '0x01 0x02 0x03 0x00 0x04 0x05 0x00' and output is called after each delimeter input")
+      THEN("The input method reports ok status for all the non-delimeter bytes")
+      THEN("The input method reports output_ready status for the delimeter byte")
+      THEN("The input_overwritten flag is false")
+      THEN("The output method reports ok status")
+      THEN("The output buffer is as expected '\x01\x02\x03' ")
+      THEN("The input method reports ok status for the subsequent non-delimeter bytes")
+      THEN("The input_overwritten flag is unchanged")
+      THEN("The input method reports output_ready status for the delimeter byte")
+      THEN("the output method reports ok status after the last delimeter byte")
+      THEN("The output buffer is as expected '\x04\x05' ")   
+
+    WHEN("input to the chunksplitter is '0x01 0x02 0x03 0x00 0x04 0x05 0x00' and output is called after all the bytes are passed")
+      THEN("The input method reports ok status on the first 3 bytes")
+      THEN("The input method reports output_ready status on the 4th byte") // delimeter byte
+      THEN("The input method reports ok status and sets the input_overwritten flag on the 5th byte")
+      THEN("The input method reports ok status for the subsequent byte and input_overwritten flag remains unchanged")
+      THEN("The output method reports ok status")
+      THEN("The output buffer only contains the last chunk '0x04 0x05' ")
+
+  GIVEN("A uint8_t ChunkSplitter constructed with default parameters, and capacity of 256 bytes which has received 256 non-delimiter bytes as input, and then a delimeter is passed as input")
+
+    WHEN("10 bytes of non-delimiter bytes followed by a delimiter byte are passed as input, and output is called between each input")
+      THEN("The input_overwritten flag is set to true on the first byte")
+      THEN("The input method reports ok status for all the non-delimeter bytes")
+      THEN("The input_overwritten flag remains unchanged after input of all 10 bytes")
+      THEN("The output method reports waiting status")
+      THEN("The input method reports output_ready status on passing the delimeter")
+      THEN("The output method reports ok status")
+      THEN("The output buffer has an expected sequence of 10 bytes without delimeters")
+
+  GIVEN("A uint8_t ChunkSplitter constructed with default parameters, and capacity of 256 bytes which is partially filled with non-delimeter input data")
+
+    WHEN("10 bytes of non-delimiter bytes followed by a delimiter byte are passed as input, and output is called between each input")
+      THEN("The input_overwritten flag is false")
```

Comment:
Since this assertion is made before any method calls, do you mean for the statement in this THEN to be part of the WHEN? (e.g. WHEN 10 bytes of non-delimiter bytes followed by a delimiter byte are passed as input, with the input method's input_overwritten flag initialized as false, and output is called...)